### ISSUE TITLE
GoPro 4 Session

### ISSUE THREAD

**Issue by thoschu96** [mapillary/mapillary_issues#1731]
Hey,

i try to use my Hero4 Session on the Iphone App.
The App is recocnizion, that there is an GoPro. i can see in the GoPro , it takes a Pictures,
but the App is Stuck on "Refreshing Preview".. nothings happenz.

Would be nice, if there would be a SKIP - Preview to start the Recording..
Or a manualy button to check Preview if Required.

Maybe it's an issue with my Memorycard, as it's showes that i have still 33339 pictures that i can take.. maybe the app is getting confused with that large amount ..
All other information like Batterielife is working .

Cheers
Thorsten

**Comment by jesolem** [comment]
Interesting. I thought the APIs didn't work on the Session. Looks like they might?

**Comment by thoschu96** [comment]
Every 10nd attempt the Preview is delivered to the App, then i can start..
But i guessing, when i would able to disable the Preview, then i might work well.
Maybe you can implement that, in the next App-Version.

I got it once passing the Preview and taking a Session.. but never uploaded them,
as it was a test-drive.

If i can help in any way, for testing with the GoPro4 Session, let me know.

**Comment by millenbop** [comment]
According to the spec the Session isn't supported, but it sounds like it is supported after all. It might be a bit slower though, that's probably why the refresh isn't working as it's using a timer and it tries to download the image while it is still processing. The GoPro camera can only do one thing at a time. I should add a cancel button anyway, thanks for reminding me.

I think I need to get access to a Session and do some tests. It's no guarantee it will work though as GoPro themselves say it isn't supported so it may stop to work in the future at any point.

Buuut, you can control the Session from the app right, and do "everything" from it? @thoschu96

**Comment by thoschu96** [comment]
an cancel button and/or an switch for "take preview".. maybe the GoPro get confused, when ít takes the first picture for preview and the Application cancels the Process and it not grabbing the picutres, as expected.

I can control the Session from the GoPro app on IOS.. no problem and LivePreview is working there.
All is working there, taking pictures, taking videos, changing setting

**Comment by millenbop** [comment]
Sounds promising. So I guess it's just some timers that needs to be adjusted. I have actually increased the timers in the version that's in review, so give it a go in the new version when it's approved, maybe that's enough.

**Comment by thoschu96** [comment]
Hey.. Times adjustment was helping.. Just installed the Update and now it's working ...

After the Capturing it's asking if i wanna upload it to the phone.. i choosed NO.
Later on, i switched to the Upload tab and was thinkig, it would automaticly connect to my GoPro
to download Images.. Maybe it's only working, when WiFi is on .. Will try it later, as i have no WiFi
available yet.

**Comment by millenbop** [comment]
So everything seems to be working? Cool!

When you go to the Uploads tab, first the app checks if there are files that needs to be synch, then it looks for external cameras. If it finds a camera, you will get a popup asking if you want to transfer images. So the phone needs to be connected to the camera, same as for taking pictures.

**Comment by thoschu96** [comment]
Yes, Its working well.

Took 2 Capturings.. One in the morning, one in the afternoon..
only the last capturing get recognized from the app..
The other Capturing with ~300 pictures is not moving into the app.

hmmmm

**Comment by millenbop** [comment]
Which version of the app did you use? We changed file formats in 4.1.4, so images taken before that will not transfer after update.

**Comment by thoschu96** [comment]
I used 4.1.5 and it was a same day capture trip.
One in the morning, one in the evening.
It looks like, that the App only remembers only one Catureingsession.


Von: Anders Mårtensson [mailto:notifications@github.com]
Gesendet: Freitag, 11. März 2016 16:01
An: mapillary/mapillary_issues <mapillary_issues@noreply.github.com>
Cc: Schulz, Thorsten <Thorsten.Schulz@syscon.biz>
Betreff: Re: [mapillary_issues] GoPro 4 Session (#1731)


Which version of the app did you use? We changed file formats in 4.1.4, so images taken before that will not transfer after update.

**Comment by millenbop** [comment]
I need some debugging info from you.

Could you connect the phone to your computer and go to iTunes. Then select your phone, then apps, then the Mapillary app, and finally copy the folder named "gopro". Please zip that and send that to me. Should be about a few Mb.

The "same" on GoPro can be done by connecting the computer to the GoPro wifi, then in a web browser type `http://10.5.5.9/gp/gpMediaList`. What you'll get is a very long JSON file of the file system on the GoPro, please send that to me as well.

You can either attach the files here or email them to me at anders@mapillary.com.

**Comment by thoschu96** [comment]
email send with log's from IPHONE and Hero4 Session

**Comment by thoschu96** [comment]
after couple test drives.
0.5 intervall seams to be a not so good selection with iphone and goPro hero4 session.
Only the first capture at all was working.. but 1.0 intervall seams to be working.

when i upload the pictures to the iphone, it shows me the date of the transfer and not of the capturing.. so please have a look into that.. 

and .. a deletion of the files from the gopro after the transfer (and the question, if i wanna delete it) would be very welcome and helpfull.

**Comment by millenbop** [comment]
Hi!

How is the Session working in the latest version of the app?

**Comment by domdomegg** [comment]
Not iOS but on Android 6.0.1 (Samsung Galaxy S6), after connecting to GoPro Wi-Fi (GoPro app works fine with it), mapillary does not recognise camera at all (including restarting with force closing the app). I can get logs if needed

**Comment by millenbop** [comment]
The Android app doesn't support action cameras yet.

**Comment by domdomegg** [comment]
Ah ok :(

**Comment by domdomegg** [comment]
Are there plans to bring GoPro support to Android? Or at least to release the Android version's source so people could add it?

**Comment by domdomegg** [comment]
Code from here may help too: https://github.com/KonradIT/goprohero

**Comment by millenbop** [comment]
Action camera support for Android IS coming, I just don't know when, sorry.

There are now official SDKs available for GoPro over at https://developer.gopro.com/

**Comment by habi** [comment]
Just my two cents: My GoPro Hero4 Session works just fine with Mapillary on iOS.
Here's an example, a bit late in the evening: https://www.mapillary.com/map/im/mxRJ6hIARpxLBOQPN3tVcQ

I'm consistently able to connect to the camera by first connecting it to the GoPro app and then starting a capture in Mapillary.
There's sometimes a problem in the preview, as mentioned in this thread, but I anyways check it in the GoPro app, so it's not really an issue for me if it doesn't work in the Mapillary app.

I'm very happy to help test and debug if I can, just let me know what you need!

**Comment by millenbop** [comment]
Yeah the peview thing is just a timing issue. I probably try to download the image too fast. Moving to the official GoPro SDK will solve all these problems, it's on my TODO list.

**Comment by millenbop** [comment]
iOS now has support for Session cameras, removing tag.

**Comment by ovivoda** [comment]
_mapillary_issues is no longer an active repository._

_Let’s continue discussing features in our forum! We will import this issue as topic to the <a href="http://forum.mapillary.com ">Mapillary Forum</a> in the coming weeks. We’ll notify here once it’s been done. See you there!_
